src/tags/object/Image/Image.js

```diff
@@ -633,7 +642,11 @@ const Model = types.model({
       if (index === self.currentImage) return;
       self.currentImage = index;
       self.currentImageEntity = self.findImageEntity(index);
-      if (isFF(FF_LSDV_4583_6)) self.preloadImages();
+
+      if (self.multiImage) {
+        self.preloadImages();
+        self.disableHistoryReinit();
```

Comment:
Why is it after preloadImage? As I understand it works just because history reset happens in a next tick, but we should not rely on this. And logically it should go first.